Automattic/wp-calypso
# Add more newspack blocks fixes

Related to: #79266 

## Proposed Changes

Follow-up from https://github.com/Automattic/wp-calypso/pull/79097#issuecomment-1629648852, to add extra checks for ETK's text domains.

The new script only succeeds if all of the following are true:
1. synced blocks exist
2. dist build files exist
3. a call to eslint contains "No lint failures found for rule(s): @wordpress/i18n-text-domain"
4. phpcs linted the same number of .php files that exist in ETK
5. phpcs has a zero exit code

(This phpcs call uses a custom config which only includes the i18n rule.)

It fails otherwise.

## Proof
Proof that it catches issues with JS textdomain in CI: https://teamcity.a8c.com/buildConfiguration/calypso_WPComPlugins_EditorToolKit/10371617?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&showLog=10371617_801_752&logView=flowAware

Proof that it catches issues with PHP textdomain in CI: https://teamcity.a8c.com/buildConfiguration/calypso_WPComPlugins_EditorToolKit/10371663?buildTab=overview&hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&showLog=10371663_1390_962&logView=flowAware

## Testing Instructions
1. run `yarn distclean` then `yarn install`
1. cd to apps/editing-toolkit
7. run `./bin/verify-textdomain.sh` -- it should fail since the blocks are not synced and JS bundles don't exist.
8. run `yarn build`
9. run the script again and it should succeed
10. Create a textdomain issue in a JS file: `apps/editing-toolkit/editing-toolkit-plugin/newspack-blocks/synced-newspack-blocks/blocks/homepage-articles/edit.js`
11. run the script and it should fail
12. Create a textdomain issue in one of the synced PHP files. (Or any ETK file). You might need to add your own line including `__`.
13. run the script and it should fail
